core/injector/application_injector.hpp

```diff
@@ -662,6 +662,69 @@ namespace kagome::injector {
     return *instance;
   }
 
+  template <typename Injector>
+  sptr<libp2p::crypto::KeyPair> get_peer_keypair(const Injector &injector) {
+    static auto initialized =
+        boost::optional<sptr<libp2p::crypto::KeyPair>>(boost::none);
+
+    if (initialized) {
+      return initialized.value();
+    }
+
+    auto &app_config =
+        injector.template create<const application::AppConfiguration &>();
+    auto &crypto_provider =
+        injector.template create<const crypto::Ed25519Provider &>();
+    auto &crypto_store =
+        injector.template create<const crypto::CryptoStore &>();
+
+    if (app_config.nodeKey()) {
+      spdlog::info("Will be used LibP2P keypair from config or args");
+
+      auto provided_keypair =
+          crypto_provider.generateKeypair(app_config.nodeKey().value());
```

Comment:
Does this method generates keypair from provided private key? If so the following assert should work

```cpp
BOOST_ASSERT(provided_keypair.secret_key == app_config.nodeKey());
```